web-ui/src/components/sentiment_icon/SentimentIcon.jsx

```diff
@@ -0,0 +1,163 @@
+      sentimentIcon = <SentimentVeryDissatisfied style={{color: isSelected ? "white" : negativeColor}}/>;
+      buttonStyle = {backgroundColor: isSelected ? negativeColor : "transparent"};
+    } else if (Math.abs(sentimentScore) <= sentimentThreshold) {
+      sentimentIcon = <SentimentSatisfied style={{color: isSelected ? "white" : neutralColor}}/>;
+      buttonStyle = {backgroundColor: isSelected ? neutralColor : "transparent"};
+    } else {
+      sentimentIcon = <SentimentVerySatisfied style={{color: isSelected ? "white" : positiveColor}}/>;
+      buttonStyle = {backgroundColor: isSelected ? positiveColor : "transparent"};
+    }
+
+    return [sentimentIcon, buttonStyle];
+  }, []);
+
+  // Renders a sentiment button that is styled according to whether the popover is open
+  const getSentimentIcon = useCallback((sentimentScore) => {
+    const [sentimentIcon, buttonStyle] = getSentimentStyles(sentimentScore, !!anchorElement);
+    return (
+      <IconButton
+        onClick={(event) => handlePopoverOpen(event)}
+        style={buttonStyle}
+      >
+        {sentimentIcon}
+      </IconButton>
+    );
+  }, [anchorElement, getSentimentStyles]);
+
+  // Renders a sentiment button that is styled according to which sentiment option is selected within the popover
+  const getSentimentOption = useCallback((sentimentScore) => {
+    const [sentimentIcon, buttonStyle] = getSentimentStyles(sentimentScore, selectedSentiment === sentimentScore);
+    return (
+      <IconButton
+        onClick={() => setSelectedSentiment(sentimentScore)}
+        style={buttonStyle}
+      >
+        {sentimentIcon}
+      </IconButton>
+    );
+  }, [selectedSentiment, getSentimentStyles]);
+
+  const updateSentiment = (newSentiment) => {
+    if (typeof currentSentiment !== "number") return;  // type check
+    if (selectedSentiment === null) return;  // do not attempt update if no option is selected
+    // prevent saving updated sentiment if it is the same as before
+    if (
+      (newSentiment < -sentimentThreshold && currentSentiment < -sentimentThreshold) ||
+      (Math.abs(newSentiment) < sentimentThreshold && Math.abs(currentSentiment) < sentimentThreshold) ||
+      (newSentiment > sentimentThreshold && currentSentiment > sentimentThreshold)
+    ) {
+      setAnchorElement(null);
+      return;
+    }
+
+    setCurrentSentiment(newSentiment);
+    // TODO: Make appropriate API call to update the sentiment
+
+    setAnchorElement(null);
+  }
+
+  const handlePopoverOpen = (event) => {
+    setShowSentimentPicker(false);
+    setSelectedSentiment(null);
+    setAnchorElement(event.currentTarget);
+  }
+
+  const handlePopoverClose = () => {
+    setAnchorElement(null);
+  }
+
+  if (props.sentimentScore === undefined || currentSentiment < -1 || 1 < currentSentiment) {
+    return (
+      <IconButton disabled>
+        <NotInterested style={{color: "gray"}}/>
+      </IconButton>
+    );
+  }
+
+  return (
+    <React.Fragment>
```

Comment:
There is a shorthand way to write React Fragments that you might find useful: 
`<></> === <React.Fragment></React.Fragment>`

https://reactjs.org/docs/fragments.html